python/cudf/cudf/core/buffer.py

```diff
@@ -60,10 +60,19 @@ def __init__(self, data=None, size=None, owner=None):
     def __reduce__(self):
         return self.__class__, (self.to_host_array(),)
 
+    @property
+    def __cuda_array_interface__(self):
+        intf = {
+            "data": (self.ptr, False),
+            "shape": (self.size,),
+            "strides": (1,),
+            "typestr": "|u1",
+            "version": 0,
+        }
+        return intf
+
     def to_host_array(self):
-        return rmm.device_array_from_ptr(
-            self.ptr, nelem=self.size, dtype="int8"
-        ).copy_to_host()
+        return cuda.as_cuda_array(self).copy_to_host()
```

Comment:
Though I did stick with Numba here (as was the case before). Despite there being some difference between Numba and CuPy array creation, it seems to transferring to host significantly dominates the time taken. So it doesn't matter much which way we go. Here's a short benchmark that builds off of the previous one

```python
In [9]: %timeit numba.cuda.as_cuda_array(rdb).copy_to_host()                    
17.3 ms ± 49.7 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [10]: %timeit cupy.asnumpy(cupy.asarray(rdb))                                
17.3 ms ± 42.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Reply:
Can we switch to cupy (or ideally the new RMM functionality for copying to a host array)? I would like to move away from using Numba internally where we're not using UDFs.

Reply:
Yeah you're right. This should be consistent. Will update to CuPy.

Also agree I'd rather move this to the RMM functionality once that lands. Just was trying something here short term.

Reply:
Have switched this to CuPy now. Can switch to RMM later.